### PR TITLE
Added Null check on Representation Property

### PR DESCRIPTION
**Pull request by amitjoy**
`org.eclipse.smarthome.core.thing.type.ThingType.getRepresentationProperty()` javadoc says that representation property of this thing type can be null. Without any representation property results in NPE.

### CODE REVIEW

File: bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/AutomaticInboxProcessor.java [hunk 1]
```diff
@@ -79,13 +81,13 @@ public void receiveTypedEvent(ThingStatusInfoChangedEvent event) {
...
         if (autoIgnore) {
             String value = getRepresentationValue(result);
-            if (value != null) {
-                Thing thing = thingRegistry.stream()
-                        .filter(t -> value.equals(getRepresentationPropertyValueForThing(t))).findFirst().orElse(null);
-                if (thing != null) {
-                    logger.debug("Auto-ignoring the inbox entry for the representation value {}", value);
-                    inbox.setFlag(result.getThingUID(), DiscoveryResultFlag.IGNORED);
-                }
+            Thing thing = thingRegistry.stream()
```
Comment (tomhoefer) [inline bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/AutomaticInboxProcessor.java]: I think it is worth to write a unit test for this purpose ....

Comment (amitjoy) [inline bundles/config/org.eclipse.smarthome.config.discovery/src/main/java/org/eclipse/smarthome/config/discovery/internal/AutomaticInboxProcessor.java]: Done

**Comment by triller-telekom** [comment]
Thanks for the fix, but while you are at it anyway, how about raising a warning in the IDE for the next one who uses `ThingType.getRepresentationProperty()`?

Just add a `@Nullable` in front of the return type `String` here: https://github.com/eclipse/smarthome/blob/master/bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/type/ThingType.java#L312

**Comment by amitjoy** [comment]
@tomhoefer I have added a new unit test to check thing when it does not contain any representation property.

@triller-telekom I have added a new commit to introduce JDT null-check annotation in `ThingType#getRepresentationProperty()`

**Pull request merged**